### PR TITLE
Refactoring of GainSelector

### PR DESCRIPTION
**Pull request by watsonjj**
This PR is for the refactoring and discussion of the existing (but mostly unused) GainSelector classes.

My next PR will be dedicated to the integration of this class in the calibration chain.

I have simplified the GainSelector class in a similar way as was recently performed for the ImageExtractor class in design. It acts as a low-level class which operates on the waveforms numpy array.

This PR picks up from #683, and includes the suggestion from #818. Other related issues include: #419, #962 and #822.

### CODE REVIEW

File: ctapipe/calib/camera/gainselection.py [hunk 1]
```diff
@@ -1,173 +1,110 @@
...
-        pass
+        if waveforms.ndim == 2:  # Return if already gain selected
+            return waveforms
+        elif waveforms.ndim == 3:
+            n_channels, n_pixels, _ = waveforms.shape
+            if n_channels == 1:  # Reduce if already single channel
+                return waveforms[0]
+            else:
+                pixel_channel = self.select_channel(waveforms)
+                return waveforms[pixel_channel, np.arange(n_pixels)]
```
Comment (thomasarmstrong) [inline ctapipe/calib/camera/gainselection.py]: Just a question, why not just 'return waveforms[pixel_channel, :]?

Comment (watsonjj) [reply ctapipe/calib/camera/gainselection.py]: Unfortunately is isn't as simple to index the array like that. `waveforms[pixel_channel, :]` returns an array of shape (n_pix, n_pix, n_samples).

Comment (watsonjj) [reply ctapipe/calib/camera/gainselection.py]: (When pixel_channel is an array of n_pix itself)

Comment (thomasarmstrong) [reply ctapipe/calib/camera/gainselection.py]: Ah ok

**Comment by kosack** [comment]
I just remembered that it's also necessary to keep the mask of which channel was used as well as the gain-selected waveform. Perhaps that could be done in another PR.

**Comment by vuillaut** [comment]
> I just remembered that it's also necessary to keep the mask of which channel was used as well as the gain-selected waveform. Perhaps that could be done in another PR.

+1 

@watsonjj Are you working on a fixing PR right now?
This PR broke lstchain with no easy way to fix it without the `pixel_channel` mask.

**Pull request merged**